### PR TITLE
global quote request changes

### LINKED ISSUE #207: Global Request Quote
https://www.moleculardevices.com/quote-request

image

### PR DESCRIPTION
**Pull request by kapildhiman**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #207

Test URLs:
- Before: https://main--moleculardevices--hlxsites.hlx.page/
- After: https://quote-request--moleculardevices--hlxsites.hlx.page/quote-request

Use Cases:
- Default form submission give credit to cmp 70170000000hlRa
- Any URL with utm parameters, gclid parameters & cmp parameters will create a session cookie
- Form (RFQ+Landing pages) submission in the above session will give credit to that particular parameter. e.g. hit test URL
 https://quote-request--moleculardevices--hlxsites.hlx.page/?cmp=testcmp&utm_medium=cpc&utm_source=facebook&utm_keyword=microplate&gclid=Tester-123

and then submit https://quote-request--moleculardevices--hlxsites.hlx.page/quote-request submission under this session will go Salesforce cmp testcmp with other parameters.

### CODE REVIEW

**Comment by mhaack** [comment]
On the parameters for the site/form do we miss a product parameter? Something replacing the pid param?

File: blocks/quote-request/quote-request.js [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+/* CREATE RFQ LIST BOX */
+function createRFQListBox(listArr, checkStep, callback) {
+  const list = document.createElement('ul');
+  list.classList.add('rfq-icon-list');
+  listArr.forEach((rfq) => {
+    const listItem = document.createElement('li');
+    const listLink = document.createElement('a');
+    const listImg = document.createElement('img');
+    const listTitle = document.createElement('span');
```
Comment (mhaack) [inline blocks/quote-request/quote-request.js:16]: You could try to use the https://github.com/hlxsites/moleculardevices/blob/main/scripts/dom-helpers.js lib to build the tags, will be less lines.

File: blocks/quote-request/quote-request.js [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+  return list;
+}
+/* CREATE RFQ LIST BOX */
+
+/* CREATE PROGRESS BAR */
+function createProgessBar(val, checkStep) {
+  const progressWrapper = document.createElement('div');
+  const progressBullet = document.createElement('div');
+  const progressBar = document.createElement('div');
+  const progress = document.createElement('div');
```
Comment (mhaack) [inline blocks/quote-request/quote-request.js:53]: Some like above maybe use dom-helper.js

File: blocks/quote-request/quote-request.js [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+
+function backOneStep(stepNum) {
+  const currentTab = document.getElementById(stepNum);
+  const prevTab = currentTab.previousElementSibling;
+
+  currentTab.style.display = 'none';
+  prevTab.style.display = 'block';
+}
+
+function createBackBtn(stepNum) {
```
Comment (mhaack) [inline blocks/quote-request/quote-request.js:81]: Also this entire function can be replaced with dom-helper.js one or two lines.

File: blocks/quote-request/quote-request.js [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+  // eslint-disable-next-line
+  description.innerHTML =
+    'A team member will contact you within 24-business hours regarding your product inquiry for : <br>';
+  productName.innerHTML = `<strong>${tab}</strong>`;
+
+  const productFamily = rfqCategories
+    .filter(({ Category }) => Category.includes(tab) > 0);
+  const sfdcProductFamily = productFamily[0].ProductFamily;
+
+  const cmpValue = getCookie('cmp') ? getCookie('cmp') : '70170000000hlRa';
```
Comment (mhaack) [inline blocks/quote-request/quote-request.js:117]: Should the default here be hard coded? Or better go as block config.

File: blocks/quote-request/quote-request.js [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+    product_family__c: sfdcProductFamily,
+    product_selection__c: sfdcProductFamily,
+    product_primary_application__c: sfdcProductFamily,
+    cmp: cmpValue,
+    google_analytics_medium__c: getCookie('utm_medium') ? getCookie('utm_medium') : '',
+    google_analytics_source__c: getCookie('utm_source') ? getCookie('utm_source') : '',
+    keyword_ppc__c: getCookie('utm_keyword') ? getCookie('utm_keyword') : '',
+    gclid__c: getCookie('gclid') ? getCookie('gclid') : '',
+    product_image: 'NA',
+    return_url: `https://www.moleculardevices.com/quote-request-success?cat=${tab}&cmp=${cmpValue}&requested_qdc_discussion__c=Quote`,
```
Comment (mhaack) [inline blocks/quote-request/quote-request.js:128]: This should be relative URL

File: blocks/quote-request/quote-request.js (hunk 1 above)
Comment (kapildhiman) [reply blocks/quote-request/quote-request.js]: Code optimized with dom helper lib

File: blocks/quote-request/quote-request.js (hunk 2 above)
Comment (kapildhiman) [reply blocks/quote-request/quote-request.js]: Code optimized with dom helper lib

File: blocks/quote-request/quote-request.js (hunk 3 above)
Comment (kapildhiman) [reply blocks/quote-request/quote-request.js:63]: Code optimized with dom helper lib

File: blocks/quote-request/quote-request.js (hunk 5 above)
Comment (kapildhiman) [reply blocks/quote-request/quote-request.js]: This must be absolute as data will be posted to the salesforce web to lead form.

File: blocks/quote-request/quote-request.js (hunk 3 above)
Comment (mhaack) [reply blocks/quote-request/quote-request.js:63]: 👍 like that

File: scripts/scripts.js [hunk 6]
```diff
@@ -421,6 +421,10 @@ async function loadPage() {
   await loadLazy(document);
   loadDelayed();
 }
+const cookieParams = ['cmp', 'utm_medium', 'utm_source', 'utm_keyword', 'gclid'];
```
Comment (mhaack) [inline scripts/scripts.js:424]: One more minor thing: let's move this to the top of file where all other constants are and upper case.

**Review by mhaack — APPROVED** [review]

**Pull request merged**